calour/io.py

```diff
@@ -45,6 +58,47 @@ def _read_biom(fp, transpose=True):
     return sid, oid, data, feature_md
 
 
+def _read_openms_csv(fp, transpose=True):
+    '''Read an OpenMS bucket table csv file
+
+    Parameters
+    ----------
+    fp : str
+        file path to the biom table
+    transpose : bool (True by default)
+        Transpose the table or not. The biom table has samples in
+        column while sklearn and other packages require samples in
+        row. So you should transpose the data table.
+
+    Returns
+    -------
+    sid : list of str
+        the sample ids
+    oid : list of str
+        the feature ids
+    data : numpy array (2d) of float
+        the table
+    feature_md : pandas DataFram
+        the feature metadata (if availble in table)
+    '''
+    logger.debug('loading OpenMS bucket table %s' % fp)
+    # use the python engine as the default (c) engine throws an error
+    # a known bug in pandas (see #11166)
+    table = pd.read_csv(fp, header=0, engine='python')
+    sample_id_col = table.columns[0]
+    sid = table.columns[1:]
+    oid = table[sample_id_col].values
+    logger.info('loaded %d samples, %d observations' % (len(sid), len(oid)))
+    data = table.values[:, 1:].astype(float)
+    feature_md = pd.DataFrame(index=oid)
+
+    if transpose:
+        logger.debug('transposing table')
+        data = data.transpose()
+
+    return sid, oid, data, feature_md
```

Comment:
i think we assume there's a dataframe for feature_metadata... tests start failing if we change to None...